File: Ecoverse-API/models/EcoSpot.js
const mongoose = require('mongoose');

// Skema untuk menyimpan koordinat GPS
const GeoSchema = new mongoose.Schema({
    type: {
        type: String,
        default: "Point"
    },
    coordinates: {
        type: [Number], // [Longitude, Latitude]
        index: '2dsphere' // Indeks penting untuk query berbasis lokasi
    }
});

const EcoSpotSchema = new mongoose.Schema({
    name: {
        type: String,
        required: true
    },
    type: {
        type: String,
        enum: ['Trash Bin', 'Recycling Station', 'Litter Hotspot'],
        required: true
    },
    location: GeoSchema, // Menggunakan skema GeoJSON
    materialAccepted: {
        type: [String],
        default: ['Mixed']
    },
    currentStatus: { // Status bin (misalnya untuk IoT/Litter Hotspot)
        type: String,
        enum: ['Empty', 'Needs Cleaning', 'Full'],
        default: 'Needs Cleaning'
    }
});

module.exports = mongoose.model('EcoSpot', EcoSpotSchema);

File: eco12/lib/models/ecospot.dart
// lib/models/ecospot.dart
import 'package:latlong2/latlong.dart'; // Digunakan untuk koordinat OpenStreetMap

class EcoSpot {
  final String id;
  final String name;
  final String type; // e.g., 'Recycling Station', 'Litter Hotspot'
  final LatLng position; // Posisi GPS (Latitude, Longitude)

  EcoSpot({
    required this.id,
    required this.name,
    required this.type,
    required this.position,
  });

  // Factory constructor untuk membuat objek EcoSpot dari data JSON (dari API Express)
  factory EcoSpot.fromJson(Map<String, dynamic> json) {
    // Catatan: MongoDB GeoJSON menyimpan [Longitude, Latitude]
    final coordinates = json['location']['coordinates'];
    final longitude = coordinates[0];
    final latitude = coordinates[1];

    return EcoSpot(
      id: json['_id'] as String,
      name: json['name'] as String,
      type: json['type'] as String,
      // Menggunakan LatLng dari latlong2
      position: LatLng(latitude, longitude), 
    );
  }
}